x-pack/plugins/apm/server/lib/search_strategies/correlations/search_strategy.test.ts

```diff
@@ -0,0 +1,221 @@
+      }
+    | {
+        logspace_ranges: estypes.AggregationsMultiBucketAggregate<{
+          from: number;
+          doc_count: number;
+        }>;
+      }
+    | {
+        latency_ranges: estypes.AggregationsMultiBucketAggregate<{
+          doc_count: number;
+        }>;
+      }
+    | undefined;
+
+  // fetchTransactionDurationPercentiles
+  if (req?.body?.aggs?.transaction_duration_percentiles !== undefined) {
+    aggregations = { transaction_duration_percentiles: { values: {} } };
+  }
+
+  // fetchTransactionDurationHistogramInterval
+  if (
+    req?.body?.aggs?.transaction_duration_min !== undefined &&
+    req?.body?.aggs?.transaction_duration_max !== undefined
+  ) {
+    aggregations = {
+      transaction_duration_min: { value: 0 },
+      transaction_duration_max: { value: 1234 },
+    };
+  }
+
+  // fetchTransactionDurationCorrelation
+  if (req?.body?.aggs?.logspace_ranges !== undefined) {
+    aggregations = { logspace_ranges: { buckets: [] } };
+  }
+
+  // fetchTransactionDurationFractions
+  if (req?.body?.aggs?.latency_ranges !== undefined) {
+    aggregations = { latency_ranges: { buckets: [] } };
+  }
+
+  return {
+    body: {
+      _shards: {
+        failed: 0,
+        successful: 1,
+        total: 1,
+      },
+      took: 162,
+      timed_out: false,
+      hits: {
+        hits: [],
+        total: {
+          value: 0,
+          relation: 'eq',
+        },
+      },
+      ...(aggregations !== undefined ? { aggregations } : {}),
+    },
+  };
+};
+
+describe('APM Correlations search strategy', () => {
+  describe('strategy interface', () => {
+    it('returns a custom search strategy with a `search` and `cancel` function', async () => {
+      const searchStrategy = await apmCorrelationsSearchStrategyProvider();
+      expect(typeof searchStrategy.search).toBe('function');
+      expect(typeof searchStrategy.cancel).toBe('function');
+    });
+  });
+
+  describe('search()', () => {
+    let mockClientFieldCaps: jest.Mock;
+    let mockClientSearch: jest.Mock;
+    let mockDeps: SearchStrategyDependencies;
+    let params: Required<PartialSearchRequest>['params'];
+
+    beforeEach(() => {
+      mockClientFieldCaps = jest.fn(clientFieldCapsMock);
+      mockClientSearch = jest.fn(clientSearchMock);
+      mockDeps = ({
+        esClient: {
+          asCurrentUser: {
+            fieldCaps: mockClientFieldCaps,
+            search: mockClientSearch,
+          },
+        },
+      } as unknown) as SearchStrategyDependencies;
+      params = {
+        index: 'apm-*',
+      };
+    });
+
+    describe('async functionality', () => {
+      it('throws an error if no params are provided', async () => {
+        const searchStrategy = await apmCorrelationsSearchStrategyProvider();
+        expect(() => searchStrategy.search({}, {}, mockDeps)).toThrow(
+          'Invalid request parameters.'
+        );
+      });
+
+      it('performs a client search with params when no ID is provided', async () => {
```

Comment:
I usually break these up if there's a `when` in there:

```suggestion
      describe('when no ID is provided', () => {
        it('performs a client search with params', async () => {
```

Reply:
Updated in 44d5cc5.